indexer/app/lib/periodic_indexer.rb

```diff
@@ -250,6 +252,11 @@ def index_round_complete(repository)
     # Give subclasses a place to hang custom behavior.
   end
 
+  def delete_pui_only_documents(updated_repositories)
```

Comment:
Is there any reason we shouldn't generalize this hook? I'm not sure the periodic indexer should reference any specific subclass as it shouldn't need to know about them, and it might also end up being useful in other cases so a more general name might be called for. Maybe something like:

```suggestion
  def repositories_updated_action(updated_repositories)
```

Reply:
Fine by me.